code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -215,28 +215,30 @@
 					var/turf/T = get_turf(src)
 					new material_drop(T)
 					qdel(src)
+					return 0
 		else if(user.drop_item())
 			W.Move(loc)
-	else
-		if(istype(W, /obj/item/stack/packageWrap))
+			return 1
+	else if(istype(W, /obj/item/weapon/weldingtool) && can_weld_shut)
+		var/obj/item/weapon/weldingtool/WT = W
+		if(!WT.remove_fuel(0, user))
 			return
-		else if(istype(W, /obj/item/weapon/weldingtool) && can_weld_shut)
-			var/obj/item/weapon/weldingtool/WT = W
-			if(!WT.remove_fuel(0, user))
+		user << "<span class='notice'>You begin [welded ? "unwelding":"welding"] \the [src]...</span>"
+		playsound(loc, 'sound/items/Welder2.ogg', 40, 1)
+		if(do_after(user, 40/WT.toolspeed, 1, target = src))
+			if(opened || !WT.isOn())
 				return
-			user << "<span class='notice'>You begin [welded ? "unwelding":"welding"] \the [src]...</span>"
-			playsound(loc, 'sound/items/Welder2.ogg', 40, 1)
-			if(do_after(user, 40/WT.toolspeed, 1, target = src))
-				if(opened || !WT.isOn())
-					return
-				playsound(loc, 'sound/items/welder.ogg', 50, 1)
-				welded = !welded
-				visible_message("<span class='notice'>[user] [welded ? "welds shut" : "unweldeds"] \the [src].</span>",
-								"<span class='notice'>You [welded ? "weld" : "unwelded"] \the [src] with \the [WT].</span>",
-								"<span class='italics'>You hear welding.</span>")
-				update_icon()
-		else
-			togglelock(user)
+			playsound(loc, 'sound/items/welder.ogg', 50, 1)
+			welded = !welded
+			visible_message("<span class='notice'>[user] [welded ? "welds shut" : "unweldeds"] \the [src].</span>",
+							"<span class='notice'>You [welded ? "weld" : "unwelded"] \the [src] with \the [WT].</span>",
+							"<span class='italics'>You hear welding.</span>")
+			update_icon()
+	else if(user.a_intent != "harm" && !(W.flags & NOBLUDGEON))
+		togglelock(user)
+		return 1
+	else
+		return ..()

```

Comment:
You can attack closed closets when on harm intent (it doesn't hurt the closet yet).